### PR TITLE
fix: 🤔 v2 breaking changes

### LINKED ISSUE #333: fix: 🤔 Wrapper do not provide all properties from the component
# Checklist (for the issue creator)

- [ ] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [ ] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [ ] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: all

# Component

The issue arises when using the following component(s): possibly all

# Environment

The issue occurred with the following operating system(s): -
The issue occurred in the following browser(s): -

# Description

The wrapper do not provide all properties from the components. All properties which do not have the `@property` tag but only a `get` and / or `set` they do not appear in the wrapper.

image

E.g. the syn-input has the `valueAsNumber` property, it appears in the custom-elements.json, but none of the wrappers (vue, angular, react) provide it.

 

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] Add documentation to wrapper docs how to use set/get properties

### LINKED ISSUE #407: fix: 🤔 add "Menu" in syn-prio-nav to translatable strings and deprecate `priority-menu-label`
# Component

`syn-prio-nav`

# Description

Currently the word `menu` is not part of the translation files. Therefore everyone would have to translate theirself by adding a prop. Instead we should use the translation files to make this translatable, so that it can switch depending on `lang=de` etc.

In addition the attribute should be deprecated, so that we can remove it in the next major release.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] syn-prio-nav uses translations (german and english)
- [ ]  `priority-menu-label` is deprecated
- [ ] deprecation info shows up in types and Storybook and informs that it will be removed in the next Major Release (@deprecated)

### LINKED ISSUE #430: fix: 🤔 fix attribute naming in syn-header
# Version

The bug occurred with the following version of the component library: 1.16.0

# Component

The issue arises when using the following component(s): syn-header

# Description

The attributes show-burger-menu and burger-menu-visible are ambiguous in its understanding.
Also they are not aligned with its figma counterparts.
Further confusion could be the fact that there are slots with the same naming but different meaning.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [x] Attributes are clear to understand (burger-menu: "opened" | "closed" | "hidden" = "hidden"
- [x] Slots are clear to understand (add "-icon" suffix for consistency reasons --> see syn-input)
- [x] Events are clear to understand
- [x] Descriptions are clear to understand
- [x] Namings should be consistent with other synergy elements
- [ ] Namings between Design and Dev are in sync

### LINKED ISSUE #434: fix: 🤔 use <slot name="label"> in syn-header instead of default slot
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [x] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [ ] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 1.16.0

# Component

The issue arises when using the following component(s) syn-header:
https://codepen.io/mariohamann/pen/wvZYxpr

As soon as you have a single space or anything in the syn-header, this is taken as "default" content and overrides the fallback. Therefore the label attribute is not taken into account.

# Description

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] Default Slot is renamed to "label"
- [ ] Documentation for default slot is renamed to "label"
- [ ] Demo-Applications are updated
- [ ] Unit-Tests are updated

### LINKED ISSUE #473: feat: ✨ remove exposed methods of angular and vue wrapper
# Checklist (for the issue creator)

- [x] I have looked at the documentation site and was not able to find the required feature.
- [x] I have looked into the Figma boards and was not able to find the required feature.
- [x] I would be willing to provide a PR for this feature.

# Component

I suggest to add the feature to the following component: angular and vue wrapper

# Story

As a developer of the Synergy Design System, I would like to remove the exposed methods from the angular and vue wrapper e.g. like the following methods of the input component `callStepUp`, `callStepDown`, `callCheckValidity`,....
These are potential places for bugs and also potential places which need to be adapted, if the web frameworks are doing updates.
If the nativeElement is exposed, we also don`t need any of these methods, because the methods can be calle don the nativeElement itself.

# Acceptance criteria

- [x] All DoD items successfully completed

- [x] adapt framework readme`s
- [x] add changes to BREAKING_CHANGES file

### LINKED ISSUE #479: feat: ✨ Change default width of open side-nav
# Checklist (for the issue creator)

- [ ] I have looked at the documentation site and was not able to find the required feature.
- [ ] I have looked into the Figma boards and was not able to find the required feature.
- [ ] I would be willing to provide a PR for this feature.

# Component

I suggest to add the feature to the following component: syn-side-nav

# Story

The default open width of the `syn-side-nav` changed in design to 320px. To be aligned and the user have the correct width without changing anything, we need to adapt the default width in the component.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] side-nav has the new default width
- [ ] AppShell templates for shrinking content use the new size per default

### LINKED ISSUE #494: fix: 🤔 Cards 'nested' property wrong meaning
# Checklist (for the issue creator)

- [x] I have verified the bug exists in the latest version of `@synergy-design-system/components` (e.g. via `pnpm install @synergy-design-system/components`).
- [x] I can provide a simple, reproducible example of the issue (e.g. via code-sandbox or sample code, ...).
- [x] I would be willing to provide a PR for this issue.

# Version

The bug occurred with the following version of the component library: 1.24.0

# Component

The issue arises when using the following component(s): `syn-card`

# Environment

The issue occurred with the following operating system(s): All
The issue occurred in the following browser(s): All

# Description

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] `nested` must be renamed to `sharp`
- [ ] Adjust docs tokens

See https://github.com/synergy-design-system/design/issues/117

### PR DESCRIPTION
**Pull request by schilchSICKAG**
# Pull Request

## 📖 Description

This PR adds breaking changes and fixes to the `<syn-header>` that are made for better usability. It also removes the `priority-menu-label` from `<syn-prio-nav>` in favor of a static translation.

It also removes the custom `call` methods exposed on our wrapper packages for angular and vue now that https://github.com/synergy-design-system/synergy-design-system/pull/478 is merged and our `nativeElement` is available on every wrapped component.

Documentation was also updated, including our first **Breaking Changes guide for Synergy** that will be shown in Storybook and the component package.

### 🎫 Issues

Closes #434
Closes #430 
Closes #407 
Closes #473
Closes #479 
Closes #333 
Closes #494 

## 👩‍💻 Reviewer Notes

- As always, check out the branch and check the functionality of `<syn-header>` and `<syn-prio-nav>`
- Have a look at the new story "Packages -> Components -> Breaking Changes" and review the texts (especially the reasoning!), overall appearance and structure.

## 📑 Test Plan

- Follow the regular procedure (e.g. running tests, looking at the demos)
- Check for broken stories
- Check for changed Chromatic Items (there should be none)

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [x] I have made sure to follow the projects coding and contribution guides.
- [ ] I have made sure that the bundle size has changed appropriately.
- [x] I have validated that there are no accessibility errors.
- [ ] ~~I have used design tokens instead of fix css values~~

### CODE REVIEW

**Comment by reichst1SICKAG** [comment]
Hi, I just tested the changes to the syn-header. 
If I set the dropdown burger-menu to opened or closed and then want to go back to hidden, the burger icon does not go away.

Tested in MacOS + Firefox

Bildschirmfoto 2024-04-30 um 15 15 13

**Comment by schilchSICKAG** [comment]
> Hi, I just tested the changes to the syn-header. If I set the dropdown burger-menu to opened or closed and then want to go back to hidden, the burger icon does not go away.

Hi @reichst1SICKAG, and thank you for testing! This seems to be a currently known issue in Storybook. As the default is set to "hidden", the docs will not update to the default value again after toggling it to something else. See #375 :(. You can see the same happening in our input when toggling the "type" from "text" (the default) to something else and trying to use "text" again.

**Comment by reichst1SICKAG** [comment]
> > Hi, I just tested the changes to the syn-header. If I set the dropdown burger-menu to opened or closed and then want to go back to hidden, the burger icon does not go away.
> 
> Hi @reichst1SICKAG, and thank you for testing! This seems to be a currently known issue in Storybook. As the default is set to "hidden", the docs will not update to the default value again after toggling it to something else. See #375 :(. You can see the same happening in our input when toggling the "type" from "text" (the default) to something else and trying to use "text" again.

Ah yes, I remember the issue. Thanks for clarifying. 👍

File: packages/components/scripts/vendorism/custom/translations.vendorism.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ const translations = {
     de: 'Menu öffnen',
     en: 'Open menu',
   },
+  priorityMenuLabel: {
```
Comment (mariohamann) [inline packages/components/scripts/vendorism/custom/translations.vendorism.js:21]: The translation should not be connected to an explicit component.

Just call the key "menu".

File: packages/components/src/components/header/header.component.ts [hunk 2]
```diff
@@ -59,46 +60,60 @@ export default class SynHeader extends SynergyElement {
...
+   * - opened: The burger menu is visible and shows the close icon
+   * - closed: The burger menu is visible and shows the open icon
    */
-  @property({ attribute: 'show-burger-menu', reflect: true, type: Boolean }) showBurgerMenu = false;
-
-  /**
-   * Determines whether or not the burger menu is currently visible.
-   */
-  @property({ attribute: 'burger-menu-visible', reflect: true, type: Boolean }) burgerMenuVisible = false;
+  @property({ attribute: 'burger-menu', reflect: true }) burgerMenu: 'hidden' | 'opened' | 'closed' = 'hidden';
```
Comment (mariohamann) [inline packages/components/src/components/header/header.component.ts:75]: I think it should be "open" to fit to others components like "syn-detail".

File: packages/components/scripts/vendorism/custom/translations.vendorism.js (hunk 1 above)
Comment (schilchSICKAG) [reply packages/components/scripts/vendorism/custom/translations.vendorism.js]: Fixed

File: packages/components/src/components/header/header.component.ts (hunk 2 above)
Comment (schilchSICKAG) [reply packages/components/src/components/header/header.component.ts]: Fixed

**Review by mariohamann — DISMISSED** [review]

**Review by kirchsuSICKAG — APPROVED** [review]

File: packages/components/BREAKING_CHANGES.md [hunk 3]
```diff
@@ -192,6 +229,231 @@ Remove `priority-menu-label` from all `<syn-prio-nav>` elements on the page. Thi
...
+**Associated Ticket(s)**:
+
+- [#479](https://github.com/synergy-design-system/synergy-design-system/issues/479)
+
+**Reason**:
+
+To better fit in case of mobile versions / smaller screen sizes, we decided to adjust the width of the opened `<syn-side-nav/>`, to be a bit narrower from 400px to 320px.
+
+**Migration Steps**:
+If for some reason the old width of 400px are desired, the css property `--side-nav-open-width` of the side navigation can be overwritten like shown in the following example.
```
Comment (mariohamann) [inline packages/components/BREAKING_CHANGES.md:245]: Do we now propagate the width somewhere? If yes, we should communicate this here.

**Review by kirchsuSICKAG — APPROVED** [review]

**Pull request merged**